app/Http/Controllers/V2/UserLocaleController.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Http\Controllers\V2;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\V2\UserLocaleRequest;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Auth;
+
+class UserLocaleController extends Controller
+{
+    public function __invoke(UserLocaleRequest $request, string $locale): JsonResponse
```

Comment:
Why is the locale coming both from the request and as a separate parameter on the controller? If we were to continue keeping it in the URL, I think `string $locale` should be dropped here, and it should be pulled from the request instead. More notes about this structure in my comment in `api_v2.php`

Reply:
Right now if you review the `UserLocaleRequest` we are not using the `locale` from the body, we are using the one on the route param.
To validate that it is one of the languages we have available.

Reply:
I now understand what you mentioned, I just obtained from the request object.